pkg/build/util/generate.go

```diff
@@ -13,6 +14,7 @@ import (
 // GenerateBuildFromConfig creates a new build based on a given BuildConfig. Optionally a SourceRevision for the new
 // build can be specified.  Also optionally a list of image names to be substituted can be supplied.  Values in the BuildConfig
 // that have a substitution provided will be replaced in the resulting Build
+// You NEED to update BuildConfig object after calling this!
```

Comment:
Ben and I talked about this and here's a rough summary (maybe he already conveyed):

The thing that sets LastChangeTrigger (LCT) info on the build config is the same one who would allocate the next build id.  It's ok if we skip.  We should ensure we save LCT and increment that build id at the same time, and that happens first before you create the build object (obviously).  The build identifier number can stay on the BuildConfig (it should always be the value that points to the most recent build, be monotonically incrementing, start at 1, etc).  There should be no more builds than number, but there may be less (if people delete, if crashes happen, etc).

Eventually, we will extend the "instantiate" control verb from podtemplates to buildconfig to reduce the client work.  I would not do that yet, because we want it to be consistent with upstream work.

----- Original Message -----
> > @@ -13,6 +14,7 @@ import (
> >  // GenerateBuildFromConfig creates a new build based on a given
> >  BuildConfig. Optionally a SourceRevision for the new
> >  // build can be specified.  Also optionally a list of image names to be
> >  substituted can be supplied.  Values in the BuildConfig
> >  // that have a substitution provided will be replaced in the resulting
> >  Build
> > +// You NEED to update BuildConfig object after calling this!
> 
> technically the same issue exists when we update the lasttriggeredid on the
> buildconfig in the imagechange_controller.  Not having transactions is
> awesome.
> 
> The way around it would be to break the sequenceid out of the builconfig
> object and have it be its own etcd key.
> 
> 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1275/files#r26386026